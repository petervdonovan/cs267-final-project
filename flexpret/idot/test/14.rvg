[mu []
[[file {Reset_Handler} {
  Reset_Handler:
    [id init]
    [[single-else get-coreid] a3
      {
        [[single-else get-hartid] t5 {}
          {
            [def (others) [memregion t0 t1 noc-base-address words 2 to 16]]
            [def (scattering) [cyclify a0 a1 others times 5]]
            [[nonzero-team pariterate-at-rate] {address} of scattering doing 2 per 19 cycles
              {
                rdcycle t3
                sw t3 [id address]
              }]
          }]
      }
      {
        [using-noc a3 [mu []
          {
            [single-hart t1
              {
                [delay a4 32]
                [iterate-at-rate t2 of [.. t4 0 to 5] doing 1 per 19 cycles
                  {
                    [noc-read t0 0]
                    [sim-print t0 t3]
                  }]
              }]
          }]]
      }]
    [single-hart t0
      {
        rdcycle t3
        [sim-print t3 t4]
      }]
    [id die]
  }]]
]
