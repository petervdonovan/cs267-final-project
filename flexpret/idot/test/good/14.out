Core-1:          0
Core-2:         26
Core-3:          9
Core-4:         26
Core-5:          3
Core-6:          0
Core-7:         10
Core-8:         20
Core-9:          0
Core-10:         14
Core-11:          9
Core-12:         14
Core-13:          3
Core-14:         20
Core-15:         10
Core-1:          0
Core-2:          8
Core-3:         20
Core-4:         30
Core-5:          3
Core-6:         22
Core-7:         10
Core-8:          8
Core-9:          0
Core-10:          0
Core-11:          9
Core-12:         22
Core-13:          3
Core-14:         30
Core-15:         10
Core-1:          0
Core-2:         21
Core-3:         22
Core-4:          1
Core-5:          0
Core-6:         20
Core-7:         20
Core-8:         18
Core-9:         14
Core-10:          7
Core-11:         26
Core-12:         19
Core-13:         14
Core-14:          6
Core-15:         26
Core-1:          0
Core-2:         27
Core-3:         22
Core-4:         24
Core-5:          2
Core-6:          3
Core-7:         20
Core-8:         25
Core-9:          9
Core-10:          2
Core-11:         31
Core-12:          1
Core-13:         11
Core-14:         26
Core-15:         29
Core-0 is done
Core-1 is done
Core-2 is done
Core-3 is done
Core-4 is done
Core-5 is done
Core-6 is done
Core-7 is done
Core-8 is done
Core-9 is done
Core-10 is done
Core-11 is done
Core-12 is done
Core-13 is done
Core-14 is done
Core-15 is done
All cores are done terminating
Assertion failed: Program terminated sucessfully
